### ISSUE TITLE
Small dams are too visually prominent at z13

### ISSUE THREAD

**Issue by Adamant36** [openstreetmap-carto/openstreetmap-carto#4231]
As of right now all dams are rendered at the same beginning zoom level. While it works for larger cement dams attached to large bodies of water, it looks weird and cluttered for smaller earthen dams that are connected to ponds. Especially if a bunch of them are mapped in the same area. Currently waterway=dam has 1,829 instances where it also has the material tag. My suggestion is to keep cement dams rendered at the current level, but start the rendering of ones tagged with material=dirt at either z14 or z15. Z15 would be best IMO, but either would be an improvement. Aside from looking better, it would also encourage people to use the material tag more. 

![Dams](https://user-images.githubusercontent.com/30259065/96964063-a5076880-14be-11eb-8a7f-626d3aba97aa.png)

**Comment by jeisenbe** [comment]
Do you have a link to the area in the example rendering?

**Comment by Adamant36** [comment]
https://www.openstreetmap.org/#map=13/40.1236/-122.4254

**Comment by imagico** [comment]
Rendering of dams is something that can definitely be improved on - though this is tricky due to technical constraints and the lack of established mapping practice for additional details on them.

The suggestion you made however is not going to work i think:

* material is tagged on dams only in about 1 percent of cases and `dirt` is definitely not a sensitive value.
* the inner structure of earth/rock fill dams is not practically verifiable and hence `surface` tagging is almost as common as `material` tagging.  Interpreting `material` would essentially incentivize adding guessed values or tagging for the renderer.
* If at all there is a relationship between size of dams and material it is that the largest dams are usually earth/rock fill dams (which is the opposite of what you seem to assume).

I would not mind in principle deciding rendering of dams based on intersecting waterways or way_area of intersecting reservoir water polygon so if you want to re-purpose this issue for that would be fine with me.  This would however require a join query which adds complexity and possible performance issues - which would need to be weighed carefully against the benefits.  Tuning the styling might be a better option.

Otherwise i would suggest to close this - using `material` for starting zoom level selection definitely does not work for me.

**Comment by jeisenbe** [comment]
Perhaps part of the visual issue here is the strong line weight and dark color currently used for dams mapped as linear ways? 

The `@dam-line` color is `#444444`
Compare to `@dam` `#adadad` (used for the fill color for dams mapped as areas)
`@weir-line` `#aaa` (And this is rendered as a series of 2x2 pixel dots rather than a solid line)

There is barely any difference between `#adadad` and `#aaa`, so why not switch `@dam` and `@dam-line` to use `#aaa` like weirs and lock gates?

**Comment by jeisenbe** [comment]
I’ve changed to title to "Small dams are too visually prominent at z13” to better explain the cartographic problem, and so we can continue to discuss different possible solutions in this issue.

**Comment by jeisenbe** [comment]
If we did want to distinguish concrete and other solid materials from earth/rock fill dams, we could use a thinner, less heavy outline for those mapped as areas, but as mentioned above, less than 1% of dams currently are tagged with this information so it is not yet a rendering priority.

**Comment by leijurv** [comment]
Adding an example: https://www.openstreetmap.org/#map=13/37.72558/-122.41224
This dam has a thick black outline only at z13, no other zoom level

**Comment by imagico** [comment]
Feature in question seems https://www.openstreetmap.org/way/1176005571

That is not well visible at z14+ because it is directly next to roads that overlap it at higher zoom levels.

**Comment by HolgerJeromin** [comment]
> Feature in question seems

You mean
https://www.openstreetmap.org/way/488659720

**Comment by dch0ph** [comment]
@leijurv Yes, the dam design doesn't work very well. I can see how it has been chosen - using 2 px wide line for both dams as linear features and areas, but the combination of a thick and dark lines is far too heavy, e.g. from a recent snapshot at Z13:

<img width="332" height="226" alt="Image" src="https://github.com/user-attachments/assets/fa9690f6-4395-4c55-9e6d-e12cba5527e4" />

The two dams and left and SE corner stick out. 

It shouldn't be too difficult to come up with a design that is less obtrusive and scales better with zoom level. A good first design issue perhaps?
